Source/Models/HTTPResponse.swift

```diff
@@ -0,0 +1,12 @@
+import Foundation
+
+public struct HTTPResponse {
+  public let data: NSData?
+  public let code: Int
+
+  public init(data: NSData?, response: NSURLResponse?) {
+    self.data = data
+    self.code = (response as? NSHTTPURLResponse)?.statusCode ?? 500
```

Comment:
Is swift able to infer the type here so you don't have to `as? NSHTTPURLResponse` it?